probability.ipynb

```diff
@@ -2966,6 +2937,34 @@
     "elimination_ask('Burglary', dict(JohnCalls=True, MaryCalls=True), burglary).show_approx()"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Elimination ask optimizations\n",
+    "\n",
+    "`elimination_ask` has some critical point to consider and some optimizations could be performed:\n",
+    "\n",
+    "- **Operation on factors**:\n",
+    "\n",
+    "  `sum_out` and `pointwise_product` function used in `elimination_ask` is where space and time complexity arise in the variable elimination algorithm (AIMA3e pg. 526).\n",
+    "\n",
+    ">The only trick is to notice that any factor that does not depend on the variable to be summed out can be moved outside the summation.\n",
+    "\n",
+    "- **Variable ordering**:\n",
+    "\n",
+    "  Elimination ordering is important, every choice of ordering yields a valid algorithm, but different orderings cause different intermediate factors to be generated during the calculation (AIMA3e pg. 527). In this case the algorithm apply a reversed order.\n",
```

Comment:
"In this case the algorithm apply a reversed order" Here, is 'algorithm':

a) plural: "In this case the algorithms apply a reversed order"

b) singular: "In this case the algorithm applies a reversed order"